src/main/java/com/zs/exercise2/Dictionary.java

```diff
@@ -0,0 +1,219 @@
+package main.java.com.zs.exercise2;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Dictionary {
+
+    static final int ALPHABET_SIZE = 26;
+
+    static class Node {
+        Node child[] = new Node[ALPHABET_SIZE];
+        boolean isEndOfWord;
+        String meaning;
+
+        Node() {
+            isEndOfWord = false;
+            meaning = "";
+            for (int i = 0; i < ALPHABET_SIZE; i++) {
+                child[i] = null;
+            }
+        }
+    };
+
+    static Node head;
+
+    static void insert(String word , String meaning) {
+        int length = word.length();
+        Node node = head;
+
+        for(int i=0;i<length;i++) {
+            char ch = word.charAt(i);
+
+            if(node.child[ch-'a'] == null) {
+                node.child[ch-'a'] = new Node();
+            }
+            node = node.child[ch-'a'];
+        }
+        node.isEndOfWord = true;
+        node.meaning = meaning;
+    }
+
+    static boolean search(String word) {
+        Node node = head;
+        int  length = word.length();
+
+        for(int i=0;i<length;i++) {
+            char ch = word.charAt(i);
+
+            if(node.child[ch-'a'] == null) {
+                return false;
+            }
+            node = node.child[ch-'a'];
+        }
+        return node.isEndOfWord;
+    }
+
+    static String meaning(String word){
+        Node node = head;
+        int  length = word.length();
+
+        for(int i=0;i<length;i++) {
+            char ch = word.charAt(i);
+
+            if(node.child[ch-'a'] == null) {
+                return null;
+            }
+            node = node.child[ch-'a'];
+        }
+        return node.meaning;
+    }
+
+
+    static ArrayList<String> correctWord(String word){
+        int length = word.length();
+
+        ArrayList<String> correct = new ArrayList<String>();
+
+        for(int i=0;i<length;i++) {
+            char ch = word.charAt(i);
+
+            for(int j=0;j<ALPHABET_SIZE;j++){
```

Comment:
instead of executing this nested loop n*26 times can you think of some better approach which will take less time.